scripts/can_errgo/error_tui.go

```diff
@@ -0,0 +1,668 @@
+
+	// True Width = 60 + 2(padding) * (# cols) = 66
+	columns := []table.Column{
+		{Title: "Error:", Width: 40},
+		{Title: "Count:", Width: 10},
+		{Title: "Recency", Width: 10},
+
+	}
+	rows := []table.Row {}
+	
+
+	t := table.New(
+		table.WithColumns(columns),
+		table.WithRows(rows),
+		table.WithFocused(true),
+		// The header will take up 1 row. If we want N rows, we need to set the height to N+1
+		table.WithHeight(11),
+	)
+
+	s := table.DefaultStyles()
+	s.Header = s.Header.	
+		BorderStyle(lipgloss.NormalBorder()).
+		BorderForeground(lipgloss.Color("240")).
+		BorderBottom(true).
+		Bold(false)
+	s.Selected = s.Selected.
+		Foreground(lipgloss.Color("229")).
+		Background(lipgloss.Color("57")).
+		Bold(false)
+	t.SetStyles(s)
+	return t
+}
+
+// Initialization of the submenu
+func createSubMenu() table.Model {
+	// Submenu has 1 column meaning it's true width is 60 + 2(padding) = 62, we need to account for the padding of the main table (66), so we add 4.
+    columns := []table.Column{
+        {Title: "Ignored Errors:", Width: 60 + 4} ,
+    }
+
+    rows := []table.Row{}
+	t := table.New(
+		table.WithColumns(columns),
+		table.WithRows(rows),
+		table.WithFocused(true),
+		// The header will take up 1 row. If we want N rows, we need to set the height to N+1
+		table.WithHeight(5),
+	)
+
+	s := table.DefaultStyles()
+	s.Header = s.Header.	
+		BorderStyle(lipgloss.NormalBorder()).
+		BorderForeground(lipgloss.Color("240")).
+		BorderBottom(true).
+		Bold(false)
+	s.Selected = s.Selected.
+		Foreground(lipgloss.Color("229")).
+		Background(lipgloss.Color("57")).
+		Bold(false)
+	t.SetStyles(s)
+
+    return t
+}
+
+
+func can_listener(p *tea.Program) {
+	// Create a raw CAN socket using the unix package
+	var sock int;
+	var err error;
+	
+	sock, err = unix.Socket(unix.AF_CAN, unix.SOCK_RAW, unix.CAN_RAW)
+	if err != nil {
+		log.Fatalf("Error creating CAN socket: %v", err)
+	}
+	defer unix.Close(sock)
+
+	// Get the index of the interface (vcan0)
+	ifi, err := net.InterfaceByName(canInterface)
+	if err != nil {
+		log.Fatalf("Error getting interface index: %v", err)
+	}
+
+	// Bind the socket to the CAN interface
+	addr := &unix.SockaddrCAN{Ifindex: ifi.Index}
+	if err := unix.Bind(sock, addr); err != nil {
+		log.Fatalf("Error binding CAN socket: %v", err)
+	}
+
+	// Create a buffer to hold incoming CAN frames
+	buf := make([]byte, canFrameSize)
+
+	fmt.Println("Listening for CAN messages on", canInterface)
+
+	for {
+		// Receive CAN message
+		_, err := unix.Read(sock, buf)
+		if err != nil {
+			log.Printf("Error receiving CAN message: %v", err)
+			continue
+		}
+
+		// Extract CAN ID and data
+		canID := binary.LittleEndian.Uint32(buf[0:4]) & 0x1FFFFFFF // 29-bit CAN ID (masked)
+		data := bytesToUint64(buf[8:]) 
+		
+		msg := CANMsg{ ID : canID, Value: data}
+		p.Send(msg)
+	}
+}
+
+
+func main() {
+	// Define the command line flag
+	warnFlag := flag.String("w", "", "warning time for the table (integer value)")
```

Comment:
The purpose of the flag package is to handle default arguments and type parsing. You should be able to remove the whole `if warnFlag else` block

Reply:
Removed